util/src/main/java/net/automatalib/util/ts/modal/MCUtil.java

```diff
@@ -0,0 +1,421 @@
+package net.automatalib.util.ts.modal;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import net.automatalib.automata.Automaton;
+import net.automatalib.automata.AutomatonCreator;
+import net.automatalib.automata.fsa.DFA;
+import net.automatalib.automata.fsa.impl.compact.CompactDFA;
+import net.automatalib.commons.util.Pair;
+import net.automatalib.commons.util.mappings.Mapping;
+import net.automatalib.serialization.dot.DOTParsers;
+import net.automatalib.serialization.dot.GraphDOT;
+import net.automatalib.ts.modal.CompactMC;
+import net.automatalib.ts.modal.CompactMTS;
+import net.automatalib.ts.modal.MTSTransition;
+import net.automatalib.ts.modal.MembershipMC;
+import net.automatalib.ts.modal.ModalContract;
+import net.automatalib.ts.modal.ModalTransitionSystem;
+import net.automatalib.ts.modal.MutableModalContract;
+import net.automatalib.ts.modal.MutableModalTransitionSystem;
+import net.automatalib.ts.modal.transitions.ModalContractEdgeProperty;
+import net.automatalib.ts.modal.transitions.ModalContractMembershipEdgePropertyImpl;
+import net.automatalib.ts.modal.transitions.ModalEdgeProperty;
+import net.automatalib.ts.modal.transitions.ModalEdgePropertyImpl;
+import net.automatalib.ts.modal.transitions.MutableModalContractEdgeProperty;
+import net.automatalib.ts.modal.transitions.MutableModalEdgeProperty;
+import net.automatalib.ts.modal.transitions.TauEdge;
+import net.automatalib.util.automata.copy.AutomatonCopyMethod;
+import net.automatalib.util.automata.copy.AutomatonLowLevelCopy;
+import net.automatalib.util.automata.fsa.DFAs;
+import net.automatalib.util.fixedpoint.Closures;
+import net.automatalib.words.Alphabet;
+import net.automatalib.words.impl.Alphabets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MCUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SystemComponent.class);
+
+    private MCUtil() {
+        // prevent instantiation
+    }
+
+    public static CompactMC<String> loadMCFromPath(String path) throws IOException {
```

Comment:
As discussed, some of the file-loading code may be better located in your application code.